src/protocol/clock/period.rs

```diff
@@ -0,0 +1,108 @@
+use std::num::{IntErrorKind, TryFromIntError};
+use std::time::Duration;
+
+pub type PeriodLength = Duration;
+pub type PeriodCount = u64;
+pub type PeriodTotalTime = Duration;
+
+pub trait Periods: Sized + Default {
```

Comment:
Is this trait really necessary? I would prefer directly implementing these functions for `Period` (unless you're thinking of a future use-case of this trait).

Reply:
Now that I have called it `Extent`, I have made it fully generic and may extracted and reused for other purposes in the future.